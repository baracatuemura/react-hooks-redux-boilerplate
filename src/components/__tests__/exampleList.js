import ExampleList from '@pageComponents/example/exampleList'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { render, fireEvent } from '@testing-library/react'

const mockStore = configureStore([])

describe('exampleList', () => {
  let component
  let mockedStore

  beforeEach(() => {
    mockedStore = mockStore({
      items: ['store item'],
    })

    component = render(
      <Provider store={mockedStore}>
        <ExampleList data-testid="teste" />
      </Provider>
    )
  })

  it('renders correctly', () => {
    expect(component.baseElement).toMatchSnapshot()
  })

  it('add an item', () => {
    mockedStore.dispatch = jest.fn()
    const { getByText, getByLabelText, findBy } = component
    const button = getByText('Add Item')
    const input = getByLabelText('Add Item ->')

    fireEvent.change(input, { target: { value: 'new item' } })
    fireEvent.click(button, new MouseEvent('click'))

    expect(mockedStore.dispatch).toHaveBeenCalledTimes(1)
  })
})
