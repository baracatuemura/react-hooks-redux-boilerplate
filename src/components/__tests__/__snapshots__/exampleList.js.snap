// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exampleList renders correctly 1`] = `
<body>
  <div>
    This is an example using all u gonna need to create any component (store, handlers, state and effects)
    <br />
    <b>
      Take a look at components/pageComponents/example/exampleList
    </b>
    <br />
    <br />
    <br />
    <h2>
      Example 1
    </h2>
    <i>
      This first 
      &lt;ul&gt;
       uses a simple reducer
    </i>
    <ul>
      <li>
        store item
      </li>
    </ul>
    <form>
      <label
        for="example-1"
      >
        Add Item -&gt;
      </label>
      <input
        id="example-1"
        type="text"
      />
      <button
        type="submit"
      >
        Add Item
      </button>
    </form>
    <br />
    <br />
    <br />
    <h2>
      Example 2
    </h2>
    <i>
      This second 
      &lt;ul&gt;
       uses an async information from store
    </i>
    <ul />
  </div>
</body>
`;
